VirtoCommerce.Storefront/Filters/ForceLoginAuthorizeFilter.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Policy;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Authorization;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace VirtoCommerce.Storefront.Filters
+{
+    /// <summary>
+    /// Authorization filter that redirects all unauthorized users to Login page (deafult AuthorizeFilter could show AccessDenied for non-authorized authenticated users)
+    /// </summary>
+    public class ForceLoginAuthorizeFilter : IAsyncAuthorizationFilter
+    {
+        public AuthorizationPolicy Policy { get; }
+
+        public ForceLoginAuthorizeFilter(AuthorizationPolicy policy)
+        {
+            Policy = policy ?? throw new ArgumentNullException(nameof(policy));
+        }
+
+        public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            // Allow Anonymous skips all authorization
+            if (context.Filters.Any(x => x is IAllowAnonymousFilter))
+            {
+                return;
+            }
+
+            var policyEvaluator = context.HttpContext.RequestServices.GetRequiredService<IPolicyEvaluator>();
```

Comment:
Try to avoid direct policy injection and use Policy from authorization service as well as in storefrontApi/account controller

Reply:
Done